ams/server/src/main/java/com/netease/arctic/server/manager/KubernetesOptimizerContainer.java

```diff
@@ -0,0 +1,155 @@
+import io.fabric8.kubernetes.client.KubernetesClientBuilder;
+import org.apache.commons.io.IOUtils;
+import org.apache.curator.shaded.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Map;
+
+/** Kubernetes Optimizer Container with Standalone Optimizer */
+public class KubernetesOptimizerContainer extends AbstractResourceContainer {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KubernetesOptimizerContainer.class);
+
+  public static final String MEMORY_PROPERTY = "memory";
+  public static final String CPU_PROPERTY = "cpu";
+  public static final String NAMESPACE = "namespace";
+  public static final String IMAGE = "image";
+  public static final String KUBE_CONFIG_PATH = "kube-config-path";
+
+  public static final String NAME = "amoro-kubernetes-optimizer";
+
+  private KubernetesClient client;
+
+  @Override
+  public void init(String name, Map<String, String> containerProperties) {
+    super.init(name, containerProperties);
+    Map<String, String> containerProperties1 = getContainerProperties();
+    // start k8s job using k8s client
+    String kubeConfigPath = checkAndGetProperty(containerProperties1, KUBE_CONFIG_PATH);
+    Config config = Config.fromKubeconfig(getKubeConfigContent(kubeConfigPath));
+    this.client = new KubernetesClientBuilder().withConfig(config).build();
+    super.init(name, containerProperties);
+  }
+
+  public static String checkAndGetProperty(Map<String, String> properties, String key) {
+    Preconditions.checkState(
+        properties != null && properties.containsKey(key), "Cannot find %s in properties", key);
+    return properties.get(key);
+  }
+
+  private String getKubeConfigContent(String path) {
+    try {
+      return IOUtils.toString(Files.newInputStream(Paths.get(path)), StandardCharsets.UTF_8);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void requestResource(Resource resource) {
+    // generate pod start args
+    long memoryPerThread =
+        Long.parseLong(checkAndGetProperty(resource.getProperties(), MEMORY_PROPERTY));
+    long memory = memoryPerThread * resource.getThreadCount();
+    // point at amoro home in docker image
+    String startUpArgs =
+        String.format(
+            "/entrypoint.sh optimizer %s %s",
+            memory, super.buildOptimizerStartupArgsString(resource));
+    LOG.info("Starting k8s optimizer using k8s client with start command : {}", startUpArgs);
+
+    String namespace = resource.getProperties().getOrDefault(NAMESPACE, "default");
+    String image = checkAndGetProperty(resource.getProperties(), IMAGE);
+    String cpu = checkAndGetProperty(resource.getProperties(), CPU_PROPERTY);
```

Comment:
Is CPU requests/limits a must? 

I think we could ask for a bit more lenient and make it optional. This will provide more flexibility in some test scenarios

Reply:
I think the first version will not introduce too complicated implementation, we may be able to unlock some advanced features when we implement PodTemplate later, the current default CPU Limit and Request is a simple implementation